tests/netdev2_test/main.c

```diff
@@ -0,0 +1,320 @@
+        puts("===========================================================");
+        puts("expected");
+        puts("===========================================================");
+        od_hex_dump(_TEST_PAYLOAD2, pkt->size, OD_WIDTH_DEFAULT);
+        puts("===========================================================");
+        puts("send data");
+        puts("===========================================================");
+        od_hex_dump(pkt->data, pkt->size, OD_WIDTH_DEFAULT);
+        return 0;
+    }
+    hdr = pkt->next;
+    /* check netif header */
+    if ((hdr->type != GNRC_NETTYPE_NETIF) || (hdr->next != NULL) ||
+        (hdr->size) != (sizeof(gnrc_netif_hdr_t) + (2 * ETHERNET_ADDR_LEN))) {
+        puts("Malformed header received");
+        return 0;
+    }
+    if (memcmp(gnrc_netif_hdr_get_src_addr(hdr->data), _test_src,
+               ETHERNET_ADDR_LEN) != 0) {
+        char addr_str[ETHERNET_ADDR_LEN * 3];
+        puts("Unexpected source received");
+        puts("=================");
+        puts("expected");
+        puts("=================");
+        puts(gnrc_netif_addr_to_str(addr_str, sizeof(addr_str),
+                                    _test_src,
+                                    ETHERNET_ADDR_LEN));
+        puts("=================");
+        puts("received source");
+        puts("=================");
+        puts(gnrc_netif_addr_to_str(addr_str, sizeof(addr_str),
+                                    gnrc_netif_hdr_get_src_addr(hdr->data),
+                                    ETHERNET_ADDR_LEN));
+        return 0;
+    }
+    if (memcmp(gnrc_netif_hdr_get_dst_addr(hdr->data), _dev_addr,
+               ETHERNET_ADDR_LEN) != 0) {
+        char addr_str[ETHERNET_ADDR_LEN * 3];
+        puts("Unexpected destination received");
+        puts("=================");
+        puts("expected");
+        puts("=================");
+        puts(gnrc_netif_addr_to_str(addr_str, sizeof(addr_str),
+                                    _dev_addr,
+                                    ETHERNET_ADDR_LEN));
+        puts("====================");
+        puts("received destination");
+        puts("====================");
+        puts(gnrc_netif_addr_to_str(addr_str, sizeof(addr_str),
+                                    gnrc_netif_hdr_get_dst_addr(hdr->data),
+                                    ETHERNET_ADDR_LEN));
+        return 0;
+    }
+
+    gnrc_pktbuf_release(pkt);
+    gnrc_netreg_unregister(GNRC_NETTYPE_UNDEF, &me);
+    return 1;
+}
+
+int main(void)
+{
+    /* initialization */
+    gnrc_pktbuf_init();
+    msg_init_queue(_main_msg_queue, _MAIN_MSG_QUEUE_SIZE);
+    netdev2_test_setup(&_dev, NULL);
+    netdev2_test_set_isr_cb(&_dev, _dev_isr);
+    netdev2_test_set_recv_cb(&_dev, _dev_recv);
+    netdev2_test_set_send_cb(&_dev, _dev_send);
+    netdev2_test_set_get_cb(&_dev, NETOPT_ADDRESS, _dev_get_addr);
+    gnrc_netdev2_eth_init(&_gnrc_dev, (netdev2_t *)(&_dev));
```

Comment:
Why do we need to initialize `netdev2_eth` here? I just want to get behind this test...